includes/class-amp-autoloader.php

```diff
@@ -96,8 +96,10 @@ class AMP_Autoloader {
 		'AMP_Content'                        => 'includes/templates/class-amp-content',
 		'AMP_Content_Sanitizer'              => 'includes/templates/class-amp-content-sanitizer',
 		'AMP_Post_Template'                  => 'includes/templates/class-amp-post-template',
+		'AMP_Carousel'                       => 'includes/utils/class-amp-carousel',
 		'AMP_DOM_Utils'                      => 'includes/utils/class-amp-dom-utils',
 		'AMP_HTML_Utils'                     => 'includes/utils/class-amp-html-utils',
+		'AMP_Image_List'                     => 'includes/utils/class-amp-image-list',
```

Comment:
As opposed to adding new AMP classes in the global namespace needing to add them to this autoloader map, we should take this as an opportunity to introduce the `AMP` namespace and then to have a PSR autoloader be able to automatically locate the class file purely based on the class name.

Reply:
What do you think about https://github.com/ampproject/amp-wp/pull/3659/commits/5794193bfc992da28eddbee4aff2fd28c4dbc322#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780R48? 

This adds a directory `includes/amp/` that can have files with the `AMP` namespace.